open_academy/models/open_academy_session.py

```diff
@@ -1,14 +1,54 @@
-from odoo import models, fields
+from odoo import models, fields, api
+from odoo.exceptions import ValidationError
 
 
 class OpenAcademySession(models.Model):
     _name = "open.academy.session"
     _description = "model of session on open academy"
 
     name = fields.Char(required=True, size=50)
-    duration = fields.Text()
-    start_date = fields.Datetime()
-    number_seats = fields.Integer()
+    duration = fields.Integer()
+    active = fields.Boolean(default=True)
+    start_date = fields.Datetime(default=fields.Date.context_today)
+    end_date = fields.Datetime()
+    number_seats = fields.Integer(required=True)
+    number_seats_taken = fields.Float(compute="_compute_number_taken_seats")
     instructor_id = fields.Many2one("res.partner", "instructor")
     course_id = fields.Many2one("open.academy.course", required=True)
     attendees_ids = fields.Many2many("res.partner", "session_res_partner_rel")
+    attendees_count = fields.Integer(compute="_compute_number_taken_seats", store=True)
+
+
+    @api.depends("number_seats", "attendees_ids")
+    def _compute_number_taken_seats(self):
+        for record in self:
+            number_attenndees = len(record.attendees_ids)
+            record.attendees_count = number_attenndees
+            if number_attenndees == 0:
+                record.number_seats_taken = 0
+                return True
+            record.number_seats_taken = (number_attenndees / record.number_seats) * 100
```

Comment:
No deberías asumir que si no hay asistentes `number_attenndees == 0` el número de asientos `number_seats` es diferente de 0. Puede ser que en la sesión nadie se ha inscrito pero cuando crearon la sesión no asignaron un número de asientos.

Si no asignan un número de asientos, esto va a fallar:
`record.number_seats_taken = (number_attenndees / record.number_seats) * 100`

Elimina el return y valida el número de asientos `number_seats` es 0, si es 0 entonces el `number_seats_taken` es 0 sino entonces haces el cálculo.

PD: Escribo en español para darme a entender mejor.

Reply:
i would did a nested condition intstead of make other comparation whit if
the nested condition make the comparation if number_seats == 0 or  number_attendees == 0 in any of those cases the compute field should display 0 so i left it like it was, nevertheless I will be on the lookout if thats its not what you asked for